.azure/publish-coverage.yml

```diff
@@ -0,0 +1,55 @@
+#
+# Copyright(c) 2021 ADLINK Technology Limited and others
+#
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License v. 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+# v. 1.0 which is available at
+# http://www.eclipse.org/org/documents/edl-v10.php.
+#
+# SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+#
+
+#
+# Azure Pipeline specifically for building and submitting to Codecov
+#
+
+trigger: none
+pr: none
+
+schedules:
+  - cron: "0 12 * * 0"
+    branches:
+      include: [ master ]
+    always: false
+
+pool:
+  vmImage: ubuntu-20.04
+
+variables:
+- group: Default
+- name: python-version
+  value: 3.9
+
+steps:
+  - template: /.azure/templates/build-cyclone.yml
+  - bash: |
+      echo "###vso[task.setvariable variable=cyclonedds_home;]$(pwd)/cyclone-$(Agent.OS)"
+    name: set_cyclonedds_home
+    displayName: Set CYCLONEDDS_HOME
+  - template: /.azure/templates/build-test.yml
+  - bash: |
+      pip install "git+https://github.com/thijsmie/codecov-python#egg=codecov"
```

Comment:
I do have a bit of a problem with the fact this references your private fork (not that Codecov has the best track-record when it comes to security). At the very least I'd like to see a ticket with an attempt being made to get these changes upstream.

Reply:
I also have a problem with it being a private fork and I have submitted a PR a while ago, see: [codecov-python!319](https://github.com/codecov/codecov-python/pull/319). Sadly there has been no response from the maintainers for a month now, I just pinged them again.

Do you want me to drop the python uploader and switch back to the bash version we use for the main repo?

Reply:
It's up to you. I'm fine with the link just being there, we can always switch back. But fyi, we don't use the bash uploader for either *cyclonedds* or *cyclonedds-cxx*. [I had a feeling just executing a downloaded script might not be the best idea](https://github.com/eclipse-cyclonedds/cyclonedds/pull/184#issuecomment-493377004).